### PR TITLE
Nbo/refactoring translation files for projects prod 13559

### PR DESCRIPTION
**Pull request by nborde-CSM**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

File: src/utils/TranslationUtils.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ const getDatasetGraphIndicatorNameTranslationKey = (graphIndicatorId) => {
 const _addResourcesToi18next = (resources) => {
   const langs = Object.keys(resources);
   langs.forEach((lang) => i18next.addResources(lang, I18N_NAMESPACE, resources[lang]));
-  i18next.reloadResources(langs);
+  // i18next.reloadResources(langs, I18N_NAMESPACE);
```
Comment (csm-thu) [inline src/utils/TranslationUtils.js:48]: Remaining commented line

File: src/services/config/i18next.js [hunk 2]
```diff
@@ -23,12 +26,27 @@ i18next
   .use(initReactI18next)
   // for all options read: https://www.i18next.com/overview/configuration-options
   .init({
-    fallbackLng: FALLBACK_LANGUAGE,
```
Comment (csm-thu) [inline src/services/config/i18next.js:26]: Shouldn't we keep the fallback language option? Removing it might break the webapp for users that don't have either EN or FR as default language.
Also, if we remove this line, then we should remove the whole declaration of `FALLBACK_LANGUAGE` in the services/configuration file

File: doc/projectBranding.md [hunk 3]
```diff
@@ -102,3 +107,137 @@ Finally, if you'd rather not display any "About" pop-up and remove the associate
...
+
+In order to edit translation labels, you can directly edit translation files :
+[public/locales/en/translation.json](../public/locales/en/translation.json) or/and
+[public/locales/fr/translation.json](../public/locales/fr/translation.json).
+
+More convenient, you can set translation custom files who override translation.json :
+[public/locales/en/custom.json](../public/locales/en/custom.json) or/and
+[public/locales/fr/custom.json](../public/locales/fr/custom.json).
+
+By default, these json files are filled with empty object
```
Comment (esasova) [inline doc/projectBranding.md:121]: I'm not sure we need to explicit it, `{}` is the default for an empty json, isn't it ?

File: doc/projectBranding.md [hunk 4]
```diff
@@ -102,3 +107,137 @@ Finally, if you'd rather not display any "About" pop-up and remove the associate
 // import { AboutContent } from '../../components/AboutContent';
 export const About = null;
 ```
+
+# Translations customization
+
+In order to edit translation labels, you can directly edit translation files :
+[public/locales/en/translation.json](../public/locales/en/translation.json) or/and
+[public/locales/fr/translation.json](../public/locales/fr/translation.json).
```
Comment (esasova) [inline doc/projectBranding.md:115]: Are we sure we want solutions to modify generic translation files ? The whole idea of _custom.json_ is to avoid it...

File: doc/projectBranding.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ description**, and **Cosmo Tech contact information**, but you can adapt its con
 will present how to configure minor changes to the default component, and how you can provide your own React component
 to replace the default one.
 
+[To customize about labels, see translation sections](#translations-customization)
```
Comment (esasova) [inline doc/projectBranding.md:11]: To make it clearer, I'd write _About_ with a capital _A_ or rephrase it to something like _To customize labels of this component._..

File: doc/projectBranding.md (hunk 4 above)
Comment (csm-thu) [reply doc/projectBranding.md]: Rephrased in a new commit :slightly_smiling_face:

File: doc/projectBranding.md (hunk 5 above)
Comment (csm-thu) [reply doc/projectBranding.md]: Rephrased in the new commit :slightly_smiling_face:

File: doc/projectBranding.md (hunk 3 above)
Comment (nborde-CSM) [reply doc/projectBranding.md]: A big refacto has been done by @csm-thu

**Review by esasova — APPROVED** [review]
LGTM :+1:

**Review by csm-thu — APPROVED** [review]
LGTM :+1:
